jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java

```diff
@@ -47,17 +60,32 @@
 public class BPMN2DataServiceImpl implements DefinitionService, DeploymentEventListener {
     
     private static final Logger logger = LoggerFactory.getLogger(BPMN2DataServiceImpl.class);
+    
     private static final BPMN2DataServiceSemanticModule MODULE = new BPMN2DataServiceSemanticModule();
+    private static final BPMN2DataServiceExtensionSemanticModule EXTENSIONS_MODULE = new BPMN2DataServiceExtensionSemanticModule(MODULE);
     
-    private ConcurrentHashMap<String, Map<String, ProcessDescRepoHelper>> definitionCache = 
+    private ConcurrentMap<String, Map<String, ProcessDescRepoHelper>> definitionCache = 
     		new ConcurrentHashMap<String, Map<String, ProcessDescRepoHelper>>();
+   
+    private static final String [] dialectNames = { "java", "mvel" };
+    private static final ProcessDialect [] dataServiceDialects = { 
+            new DataServiceJavaProcessDialect(),
+            new DataServiceMvelProcessDialect()
```

Comment:
Yes -- well, sort of. In the builders that the dialects provide, I was only planning on extracting information and skipping the actual compilation of the expressions/scripts (to save time). However, it's very easy to extend them to make sure that the expressions/scripts are compiled and added.

(So for v7, it's easy to have one class that also extracts the info, depending on a flag or whatever.. )